include/evmc/loader.h

```diff
@@ -60,15 +60,16 @@ enum evmc_loader_error_code
  *   "libexample-interpreter.so",
  * - the "lib" prefix and file extension are stripped from the name:
  *   "example-interpreter"
- * - all "-" are replaced with "_" to construct _full name_:
+ * - all "-" are replaced with "_" to construct _base name_:
  *   "example_interpreter",
- * - the _full name_ is split by "_" char and the last item is taken to form the _short name_:
- *   "interpreter",
- * - the name "evmc_create_" + _full name_ is checked in the library:
+ * - the function name "evmc_create_" + _base name_ is searched in the library:
  *   "evmc_create_example_interpreter",
- * - then, the name "evmc_create_" + _short name_ is checked in the library:
- *   "evmc_create_interpreter".
- * - lastly, the name "evmc_create" is checked in the library
+ * - if function not found, the _base name_ is shorten by skipping the first word separated by "_":
+ *   "interpreter",
+ * - then, the function of the shorter name "evmc_create_" + _base name_ is searched in the library:
+ *   "evmc_create_interpreter",
```

Comment:
So what exactly is the logic here? It removes the prefixes? `example` off `example_interpreter`

Reply:
yes

Reply:
But also `a_b_c_d` -> `b_c_d` -> `c_d` -> `d`.

Reply:
What is the reasoning for that?

Reply:
`evmc-example-vm` :) I want it to come up with function name `evmc_create_example_vm()` while previously it tried only `evmc_create_evmc_example_vm()` and `evmc_create_vm()`.

Reply:
I think that may not be the most sound reasoning :)

Why not just name the file `example-vm` (and the creator appropriately) and be done with it?

Reply:
For 3 reasons:
- I don't want to name the library example-vm in case it is going to be installed.
- This code looks better than before.
- Even the author of loader (me) was no able to remember how exactly it was working before so #116 does not work.

Reply:
It was, but not fully. It was skipping all leading words except the last one.